stl/inc/execution

```diff
@@ -137,6 +137,35 @@ template <>
 struct is_execution_policy<execution::unsequenced_policy> : true_type {};
 #endif // _HAS_CXX20
 
+template <class _Ty, class _FwdIt>
+void _Construct_in_place_by_deref(_Ty& _Val, const _FwdIt& _Iter) {
+    ::new (static_cast<void*>(_STD addressof(_Val))) _Ty(*_Iter);
+}
+
+template <class _Ty, class _UnaryOp, class _FwdIt>
+void _Construct_in_place_by_transform_deref(_Ty& _Val, _UnaryOp& _Tranform_op, const _FwdIt& _Iter) {
+    ::new (static_cast<void*>(_STD addressof(_Val))) _Ty(_Tranform_op(*_Iter));
+}
+
+template <class _Ty, class _BinaryOp, class _ArgTy>
+void _Implicitly_construct_in_place_by_binary_op(_Ty& _Val, _BinaryOp& _Reduce_op, _ArgTy& _Left, _ArgTy& _Right) {
+    ::new (static_cast<void*>(_STD addressof(_Val))) _Ty([&]() -> _Ty { return _Reduce_op(_Left, _Right); }());

```

Comment:
And capturing it by reference too. Ditto below.

Reply:
The point is performing implicit conversion in placement new with workaround for imperfectness of forwarding. So I guess copying should be avoided.